vyper/compiler/settings.py

```diff
@@ -12,3 +14,31 @@
     VYPER_TRACEBACK_LIMIT = int(_tb_limit_str)
 else:
     VYPER_TRACEBACK_LIMIT = None
+
+
+class OptimizationLevel(Enum):
+    NONE = 1
+    GAS = 2
+    CODESIZE = 3
+
+    @classmethod
+    def from_string(cls, val):
+        match val:
+            case "none":
+                return cls.NONE
+            case "gas":
+                return cls.GAS
+            case "codesize":
+                return cls.CODESIZE
+        raise ValueError(f"unrecognized optimization level: {val}")
+
+    @classmethod
+    def default(cls):
+        return cls.GAS
+
+
+@dataclass
+class Settings:
+    compiler_version: Optional[str] = None
+    optimize: Optional[OptimizationLevel] = None
+    evm_version: Optional[str] = None
```

Comment:
Could do a similar thing for evm version as a str enum

Reply:
yea i thought about this but it causes evm version to be imported very early in the compilation pipeline, evm versions are validated later anyways. i think we should consider it later as i am planning on cleaning up the `compile_codes` API when i get to working on modules.